fe/fe-core/src/main/java/com/starrocks/server/TemporaryTableMgr.java

```diff
@@ -0,0 +1,169 @@
+        TemporaryTableTable tables = tablesMap.get(sessionId);
+        tables.addTable(databaseId, tableName, tableId);
+        LOG.info("add temporary table, session[{}], db id[{}], table name[{}], table id[{}]",
+                sessionId.toString(), databaseId, tableName, tableId);
+    }
+
+    public Long getTable(UUID sessionId, long databaseId, String tableName) {
+        if (!tablesMap.containsKey(sessionId)) {
+            return null;
+        }
+        return tablesMap.get(sessionId).getTableId(databaseId, tableName);
+    }
+
+    public boolean tableExists(UUID sessionId, long databaseId, String tblName) {
+        if (!tablesMap.containsKey(sessionId)) {
+            return false;
+        }
+        return tablesMap.get(sessionId).getTableId(databaseId, tblName) != null;
+    }
+
+    public void dropTemporaryTable(UUID sessionId, long databaseId, String tableName) {
+        TemporaryTableTable tables = tablesMap.get(sessionId);
+        if (tables == null) {
+            return;
+        }
+        tables.removeTable(databaseId, tableName);
+        LOG.info("drop temporary table, session[{}], db id[{}], table name[{}]",
+                sessionId.toString(), databaseId, tableName);
+    }
+
+    public Table<Long, String, Long> getTemporaryTables(UUID sessionId) {
+        TemporaryTableTable tables = tablesMap.get(sessionId);
+        if (tables == null) {
+            return HashBasedTable.create();
+        }
+        return tables.getAllTables();
+    }
+
+    public void removeTemporaryTables(UUID sessionId) {
+        tablesMap.remove(sessionId);
+        LOG.info("remove all temporary tables in session[{}]", sessionId.toString());
+    }
+
+    public List<String> listTemporaryTables(UUID sessionId, long databaseId) {
+        TemporaryTableTable tables = tablesMap.get(sessionId);
+        if (tables == null) {
+            return Lists.newArrayList();
+        }
+        return tables.listTables(databaseId);
+    }
+
+    // get all temporary tables under specific databases, return a Table<databaseId, sessionId, tableId>
+    public Table<Long, UUID, Long> getAllTemporaryTables(Set<Long> requiredDatabaseIds) {
+        Table<Long, UUID, Long> result = HashBasedTable.create();
+        tablesMap.forEach((sessionId, tables) -> {
+            // db id -> table name -> table id
+            Table<Long, String, Long> allTables = tables.getAllTables();
+            for (Table.Cell<Long, String, Long> cell : allTables.cellSet()) {
+                if (requiredDatabaseIds.contains(cell.getRowKey())) {
+                    result.put(cell.getRowKey(), sessionId, cell.getValue());
+                }
+            }
+
+        });
+        return result;
+    }
+
+    public Set<UUID> listSessions() {
+        return tablesMap.keySet();
+    }
+
+
+    @VisibleForTesting
+    public void clear() {
+        if (tablesMap != null) {
+            tablesMap.clear();
+        }
+        System.gc();
```

Comment:
Is this necessary? And can this method be called in production env or just for testing purpose?

Reply:
> Is this necessary? And can this method be called in production env or just for testing purpose?

just for test， won't be used in production env @nshangyiming